torch/csrc/jit/codegen/cuda/ops/alias.cpp

```diff
@@ -52,6 +52,33 @@ TensorView* applyViewTransforms(
 
 } // namespace
 
+TensorView* view(TensorView* x, DataType dtype) {
+  if (x->getDataType() == dtype) {
+    return x;
+  }
+  // TODO: support view(dtype) for dtypes of different size.
+  TORCH_INTERNAL_ASSERT(
+      dataTypeSize(x->getDataType().value()) == dataTypeSize(dtype));
+
+  std::vector<IterDomain*> out_domain;
+  // TODO: should the following line be:
+  //    TensorDomain::noReductions(x->getMaybeRFactorDomain());
+  // ? What does it mean?
```

Comment:
Yes, `inp_domain` should be `TensorDomain::noReductions(x->getMaybeRFactorDomain())`.

`TensorDomain::noReductions()` just removes reduction IterDomains. 

`getMaybeRFactorDomain()` returns an rfactor domain if `x` has one. This matters, e.g., `x` is produced by a view operation, where its root domain has the same domain as the input tensor of the view op, and the rfactor domain has the domain after the view. Since this `ViewDtype` op sees `x` as its input, it should look at the post-view domain rather than the pre-view domain.

Reply:
Thanks for explaining! I have updated this part of code.